frontend/toolkit-webapp/src/pages/ResetPassword.js

```diff
@@ -0,0 +1,162 @@
+import { Button, Container, TextField, Typography, Paper, Box } from '@material-ui/core'
+import config from '../config.json'
+import loginStyles from '../styles/loginStyle'
+import TopNavBar from '../components/TopNavBar'
+import LockOutlinedIcon from '@material-ui/icons/LockOutlined';
+import Avatar from '@material-ui/core/Avatar';
+import videoSource from '../img/PEOSTRI.mp4'
+import jwt_decode from "jwt-decode";
+
+function ResetPassword(props) {
+
+    const [password, setPassword] = useState('')
+    const [passwordCopy, setPasswordCopy] = useState('')
+
+    const classes = loginStyles()
+
+    useEffect(() => {
+        const pathname = window.location.pathname.split('/') //returns the current path
+        const resetToken = pathname[pathname.length - 1]
+        
+        let res = undefined
+        checkCreds(resetToken)
+
+    }, []);
+
+    const checkCreds = async (resetToken) => {
+
+        let res = undefined
+
+        res = await fetch(config.server_url + config.paths.checkResetCreds, {
+            method: 'POST',
+            headers: {
+                'Content-type': 'application/json'
+            },
+            body: JSON.stringify({"resetToken": resetToken})
+
+        })
+
+        const data = await res.json()
+        
+        if(data.message === 'credentials disapproved')
+        {
+            alert(data.message)
+            alert('Your token may have expired. Please input your information again for a follow-up email.')
+            props.history.push('../forgot');
+        }
+
+    }
+
+    const onSubmit = (e) => {
+        e.preventDefault()
+        if (!password || !passwordCopy) {
+            alert('Please fill in both blanks')
+            return
+        }
+
+        if(password !== passwordCopy)
+        {
+            alert('Passwords must match!')
+            return
+        }
+
+        onResetPass({password})
+
+    }
+
+    const onResetPass = async (creds) => {
+
+        const pathname = window.location.pathname.split('/') //returns the current path
+        const resetToken = pathname[pathname.length - 1]
+        
+        const res = await fetch(config.server_url + config.paths.resetPassApproved, {
+            method: 'POST',
+            headers: {
+                'Content-type': 'application/json'
+            },
+            body: JSON.stringify({ "resetToken": resetToken, "password": creds.password })
+        })
+        
+        const data = await res.json()
+
+        if (data.message === "Success!") {
+            alert('Password has been changed!')
+            props.history.push('../../login')
+        }
+        else
+        {
+            alert('Something went wrong! Try inputting your new password again')
+            return
+        }
+       
+    }
+
+    const returnToLogin = (e) => {
+        props.history.push('../login');
+    }
+
+    return (
+        <div style={{height:'100vh', width:'100vw'}}>
+            <TopNavBar hideComponents={true}/>
```

Comment:
Home button icon still displaying on this page, should be removed. Clicking it re-routes user to blank dashboard page.

Reply:
I'm not sure how to fix this one, since the url is reset/{resetToken} and I'm not sure how that would be accessed in the topBar when I'm creating it in ResetPassword.

Reply:
Fixed.